stdlib.sh

```diff
@@ -654,12 +840,443 @@ function __STDLIB_API_1_info() { # {{{
+
+
+# This function may be overridden
+#
+# Warning closing tag
+#
+function __STDLIB_API_1_resWarning() { # {{{
+	__STDLIB_API_1_showRes "${std_COLOUR_START_WARNING}WARNING${std_COLOUR_END}" "${@:-}"
+
+	# Don't stomp on std_ERRNO
+	return 1
+} # __STDLIB_API_1_resWarning # }}}
+
+# This function may be overridden
+#
+# Error closing tag
+#
+function __STDLIB_API_1_resKo() { # {{{
+	__STDLIB_API_1_showRes "${std_COLOUR_START_FAIL}FAIL${std_COLOUR_END}" "${@:-}"
+
+	# Don't stomp on std_ERRNO
+	return 1
+} # __STDLIB_API_1_resKo # }}}
+
+
+# This function may be overridden
+#
+# Fatal closing tag
+#
+function __STDLIB_API_1_resFail() { # {{{
+	__STDLIB_API_1_showRes "${std_COLOUR_START_FAIL}FAIL${std_COLOUR_END}" "${@:-}"
+	die
+
+	# Dead code, die will immediately quit
+	return 1
+} # __STDLIB_API_1_resFail # }}}
+
+
+# This function may be overridden
+#
+# Logs tag, calls function, then dies
+#
+function __STDLIB_API_1_callAndDie() { # {{{
+	local functionName="${1:-}" ; shift
+	local data="${*:-}"
+
+	std_DEBUG=1 __STDLIB_API_1_std::log >&2 "[  ${std_COLOUR_START_FATAL}FATAL${std_COLOUR_END}  ]" "${data}"
+	[[ -n "${functionName}" ]]  &&  eval "${functionName}"
+	__STDLIB_API_1_std::cleanup 1
+
+	# Dead code, cleanup will call die and immediately quit
+	return 1
+} # __STDLIB_API_1_callAndDie # }}}
+
+
+# This function may be overridden
+#
+# Executes subcommands by:
+# 1. checking 'PRETEND' option
+# 2. capturing output stream, as specified
+# 3. returning ret value and exit code from subcommand to parent caller
+# Dedicated global vars are used to enhance flexibility on return values
+# handling, although this is limited to one level of nested calls
+#
+# Implementation now relies on 'capture' in order to avoid duplicated logic for
+# stream capture
+#
+function __STDLIB_API_1_evalRes() { # {{{
+	local command="${1:-${STDLIB_EXEC_COMMAND}}"
+	local -i param_selectStream=${2:-$STDLIB_STREAM_NORMAL}
+
+	STDLIB_EXEC_COMMAND="${command}"
+	STDLIB_EXEC_RESULT=""
+	STDLIB_EXEC_STATUS=0
+
+	(( $std_PRETEND )) && debug "Pretend to run '${command}'" $STDLIB_DEBUG_FUNCTION
+	(( !std_PRETEND )) && debug "About to run '${command}'" $STDLIB_DEBUG_FUNCTION
+
+	if (( !std_PRETEND )); then
+		STDLIB_EXEC_RESULT="$( __STDLIB_API_1_std::capture "$param_selectStream" "${command}" )"
+		STDLIB_EXEC_STATUS=$?
+	fi
+
+	debug "*** S:evalRes() ***" $STDLIB_DEBUG_FUNCTION
+	debug "MYSTATUS: '${STDLIB_EXEC_STATUS}'" $STDLIB_DEBUG_FUNCTION
+	debug "MYRESULT: '\n${STDLIB_EXEC_RESULT}\n'" $STDLIB_DEBUG_FUNCTION
+	debug "*** E:evalRes() ***" $STDLIB_DEBUG_FUNCTION
+
+	respond "${STDLIB_EXEC_RESULT}"
+	debug "MYSTATUS: '${STDLIB_EXEC_STATUS}'" $STDLIB_DEBUG_FUNCTION
+
+	if (( STDLIB_EXEC_STATUS )); then
+		std_ERRNO="$( errsymbol ESUBSHELL )"
+	else
+		std_ERRNO="$( errsymbol ENOERROR )"
+	fi
+
+	return $STDLIB_EXEC_STATUS
+
+} #__STDLIB_API_1_evalRes # }}}
+
+
+# This function may be overridden
+#
+# Wrapper to evalRes(), for backward compatibility
```

Comment:
With something I did not port here from my own version of stdlib.sh :P
It's called: transcendental or astral compatibility because the function transcendentally refers to another function that lives in a non-sourced file :P